model/task/task.go

```diff
@@ -3669,6 +3669,58 @@ func GetGroupedTaskStatsByVersion(versionID string, opts GetTasksByVersionOption
 
 }
 
+// GetMatchingBaseTasks returns the corresponding base tasks for tasks on a version.
+// If a task ran on the base version but was not scheduled on the version, then it won't be returned.
+func GetMatchingBaseTasks(versionID string, baseVersionID string) ([]Task, error) {
+	pipeline := []bson.M{}
+
+	pipeline = append(pipeline,
+		bson.M{"$match": bson.M{VersionKey: baseVersionID}},
+	)
+
+	// Look up and filter out tasks that don't exist on version.
+	VersionTasksKey := "VersionTasks"
```

Comment:
Since it's only used in this function it could be okay to leave here but shouldn't be an uppercase constant

Reply:
If we want to keep it here, we can rewrite it as 
`const versionTasksKey = "VersionTasks"`

Reply:
however, we do already have `dependencyKey = "dependencies"` up top, so adding it there would keep them together.

Reply:
Fair; I also don't really think that should be a top-of-file constant either though since it's just a variable to make sure the string is used consistently within a query, rather than a constant that defines some behavior. I'm not invested either way though as long as it's private.

Reply:
same